test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -306,18 +306,23 @@ public:
         return *this;
     }
 
-    // If ck_ranges is passed, verifies only that information relevant for ck_ranges matches.
-    flat_reader_assertions& produces_range_tombstone(const range_tombstone& rt_, const query::clustering_row_ranges& ck_ranges = {}) {
-        testlog.trace("Expect {}", rt_);
```

Comment:
> This is obviously fragile; 

Why is it fragile?

Reply:
Version converters may split range tombstones.  There is at last one test that uses `flat_reader_assertions::produces(const schema&, const mutation_fragment&)` on range tombstones, and without this change the later changes in this pull break it (for no good reason).